pkg/provider/ip/eni/eni.go

```diff
@@ -48,177 +49,223 @@ type eni struct {
 }
 
 type ENIManager interface {
-	InitResources(ec2APIHelper api.EC2APIHelper) ([]string, error)
-	CreateIPV4Address(required int, ec2APIHelper api.EC2APIHelper, log logr.Logger) ([]string, error)
-	DeleteIPV4Address(ipList []string, ec2APIHelper api.EC2APIHelper, log logr.Logger) ([]string, error)
+	InitResources(ec2APIHelper api.EC2APIHelper) ([]string, []string, error)
+	CreateIPV4Resource(required int, resourceType config.ResourceType, ec2APIHelper api.EC2APIHelper, log logr.Logger) ([]string, error)
+	DeleteIPV4Resource(ipList []string, resourceType config.ResourceType, ec2APIHelper api.EC2APIHelper, log logr.Logger) ([]string, error)
 }
 
 // NewENIManager returns a new ENI Manager
 func NewENIManager(instance ec2.EC2Instance) *eniManager {
 	return &eniManager{
-		ipToENIMap: map[string]*eni{},
-		instance:   instance,
+		resourceToENIMap: map[string]*eni{},
+		instance:         instance,
 	}
 }
 
-// InitResources loads the list of ENIs and IPs associated with the instance
-func (e *eniManager) InitResources(ec2APIHelper api.EC2APIHelper) ([]string, error) {
+// InitResources loads the list of ENIs, secondary IPs and prefixes, associated with the instance
+func (e *eniManager) InitResources(ec2APIHelper api.EC2APIHelper) ([]string, []string, error) {
```

Comment:
Same here, is that possible to consolidate IP and PD together into a Resource struct to be returned and handled other places as well?

Reply:
Yeah I consolidated them into a struct and updated eni API here. Will make another PR for related changes on resource providers.